ksql-engine/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.metrics;
+
+import io.confluent.common.utils.SystemTime;
+import io.confluent.common.metrics.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Topic based collectors for producer/consumer related statistics that can be mapped on to streams/tables/queries for ksql entities (Stream, Table, Query)
+ * TODO: hook into Stream, Table, Query statistics for describe
+ * TODO: persistence, bouncing a node should not blat current stats
+ * TODO: add lastMinute stats collector (max events, max bytes, timestamp)
+ * TODO: add maxMinute stats retention
+ */
+public class MetricCollectors {
+
+
+  private static final Map<String, MetricCollector> collectorMap = new HashMap<String, MetricCollector>();
+  private static final Metrics metrics;
+
+  static {
+    MetricConfig metricConfig = new MetricConfig().samples(100).timeWindow(1000, TimeUnit.MILLISECONDS);
+    List<MetricsReporter> reporters = new ArrayList<>();
+    reporters.add(new JmxReporter("ksql.metrics"));
+    metrics = new Metrics(metricConfig, reporters, new SystemTime());
+  }
+
+  public static Metrics addCollector(MetricCollector collector) {
```

Comment:
package private.
Also, is this thread-safe? Will there ever be multiple threads accessing this? I believe there will be as these will be created for each `StreamThread` i think